engine/gamesys/src/gamesys/components/comp_collision_object.cpp

```diff
@@ -688,15 +688,15 @@ namespace dmGameSystem
                 dmPhysicsDDF::ContactPoint& a = ddf.m_A;
                 a.m_Group               = group_hash_a;
                 a.m_Id                  = instance_a_id;
-                a.m_Position            = dmGameObject::GetWorldPosition(instance_a);
+                a.m_Position            = contact_point.m_PositionA;
                 a.m_Mass                = mass_a;
                 a.m_RelativeVelocity    = -contact_point.m_RelativeVelocity;
                 a.m_Normal              = -contact_point.m_Normal;
 
                 dmPhysicsDDF::ContactPoint& b = ddf.m_B;
                 b.m_Group               = group_hash_b;
                 b.m_Id                  = instance_b_id;
-                b.m_Position            = dmGameObject::GetWorldPosition(instance_b);
```

Comment:
Don't we need instance positions as well as it is in `dmPhysicsDDF::ContactPointResponse` ?

Reply:
I'm not sure.
The corresponding "response" message has
* position = contact point
* other_position = world point of other object

So it doesn't have the actual object position (only for the other object)
If I _did_ add the object position in the struct, what would the name be? object_position ?
It's not a big deal, but I went with "less is more" for now?

Reply:
using messages we send two of them to each object, that's why we use such terms like "other_position". But it's only one event for a callback.

so, yes, I think it should be `instance_position` or `object_possition`. 
Don't forget to change documentation either: 
https://github.com/defold/defold/blob/9a87526fe7ff093886e6575f2927300053dd0559/engine/gamesys/src/gamesys/scripts/script_physics.cpp#L1561-L1582

https://github.com/defold/defold/blob/9a87526fe7ff093886e6575f2927300053dd0559/engine/gamesys/proto/gamesys/physics_ddf.proto#L186-L197

Reply:
"using messages we send two of them to each object"
I was referring to this fact, that the message itself doesn't actually have its _own_ instance position, only the _other_ position.

But, for ease of use I think it's good to add a instance position to each sub struct.

Reply:
>I was referring to this fact, that the message itself doesn't actually have its own instance position, only the other position.

yes, because each of them kinda know it's own position, and only get the other position. There are two ways to solve it: provide both because even isn't in object, so it needs both or provide none (may be requested by id as I understand?)